app/src/main/java/ch/epfl/culturequest/ui/profile/ProfileViewModel.java

```diff
@@ -33,28 +33,31 @@ public ProfileViewModel(String uid) {
         username = new MutableLiveData<>();
         profilePictureUri = new MutableLiveData<>();
         pictures = new MutableLiveData<>();
-        followed = new MutableLiveData<>();
+        followed = new MutableLiveData<>(false);
 
         EspressoIdlingResource.increment();
         if (profile != null) {
-            if (selectedProfile != null && selectedProfile.getUid().equals(uid)) {
-                username.setValue(selectedProfile.getUsername());
-                profilePictureUri.setValue(selectedProfile.getProfilePicture());
-                // We load all the posts for a user in 1 query to the database. Initially, I queried only 4 posts at
-                // a time, but it is computationally more efficient to do 1 big query:
-                //https://stackoverflow.com/questions/3910317/is-it-better-to-return-one-big-query-or-a-few-smaller-ones#:~:text=It%20is%20significantly%20faster%20to,the%20server%20more%20each%20time.
-                CompletableFuture<List<Post>> profilePosts = selectedProfile.retrievePosts();
-                profilePosts.handle((posts, t) -> {
-                    if (posts != null && t == null) {
-                        pictures.setValue(posts);
-                    }
-                    return null;
+            if (!profile.getUid().equals(uid)) {
+                Database.getProfile(uid).whenComplete((selectedProfile, e) -> {
+                    this.selectedProfile.setValue(selectedProfile);
+                    username.setValue(selectedProfile.getUsername());
+                    profilePictureUri.setValue(selectedProfile.getProfilePicture());
+                    // We load all the posts for a user in 1 query to the database. Initially, I queried only 4 posts at
+                    // a time, but it is computationally more efficient to do 1 big query:
+                    //https://stackoverflow.com/questions/3910317/is-it-better-to-return-one-big-query-or-a-few-smaller-ones#:~:text=It%20is%20significantly%20faster%20to,the%20server%20more%20each%20time.
+                    CompletableFuture<List<Post>> profilePosts = Database.getPosts(selectedProfile.getUid());
+                    profilePosts.handle((posts, t) -> {
```

Comment:
Is there a reason why you don't use "getValue()" here to get the Uid of the MutableData selectedProfile ?

Reply:
yes, if you check the few lines over these ones, it calls the Database: ```Database.getProfile(uid).whenComplete((selectedProfile, e) -> { ...```
so the selected profile you mention is the one fetched from the database not the mutable live data. In that block i write `this.selectedProfile.setValue(selectedProfile);` so i could either use the mutable live data and use `getValue().getUid()` or just use the profile fetched from the database